near-sdk/src/types/gas.rs

```diff
@@ -0,0 +1,140 @@
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use core::ops;
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+/// Represents the amount of NEAR tokens in "gas units" which are used to fund transactions.
+#[derive(
+    Default,
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    PartialOrd,
+    Ord,
+    Eq,
+    BorshSerialize,
+    BorshDeserialize,
+    Hash,
+    BorshSchema,
+)]
+pub struct Gas(pub u64);
```

Comment:
I think we might also want to slap `#[repr(transparent)]` here, but we also can just wait until someone asks for it.